lib/SIL/SILType.cpp

```diff
@@ -309,15 +309,28 @@ bool SILType::canRefCast(SILType operTy, SILType resultTy, SILModule &M) {
 }
 
 SILType SILType::getFieldType(VarDecl *field, SILModule &M) const {
-  assert(field->getDeclContext() == getNominalOrBoundGenericNominal());
+  auto baseTy = getSwiftRValueType();
+
+  if (field->getDeclContext() != getNominalOrBoundGenericNominal()) {
+    // Look up the member in a superclass.
```

Comment:
How do you know the field is in a superclass and not in the protocol?

Reply:
No other nominal type has stored fields inherited from another decl.

Reply:
maybe assert that the field's DeclContext is a class context then

Reply:
also this check is wrong -- imagine if the field's DeclContext is an extension of getNominalOrBoundGenericNominal(), then you will evaluate true here and do the wrong thing.

Reply:
Extensions never have stored properties.